NGit/NGit.Patch/BinaryHunk.cs

```diff
@@ -168,7 +168,7 @@ internal virtual int ParseHunk(int ptr, int end)
 			//
 			while (ptr < end)
 			{
-				bool empty = buf[ptr] == '\n';
+				bool empty = (buf[ptr] == '\n') || (buf[ptr] == '\r' && buf[ptr-1] == '\n');
```

Comment:
This is tested by an existing test. The change we made to `Patch.Parse` (reverting an old change) causes this parsing to fail. So we fix this to ensure the tests written for the original issue still pass.